client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/hooks/use-pattern-pages.ts

```diff
@@ -1,31 +1,105 @@
 import { useSearchParams } from 'react-router-dom';
-import { INITIAL_PAGES } from '../constants';
-import type { Pattern } from '../types';
+import { INITIAL_PAGES, ORDERED_PATTERN_PAGES_CATEGORIES } from '../constants';
+import { useCategoriesOrder } from '../hooks';
+import type { Pattern, Category, CustomPageTitle } from '../types';
 
-const usePatternPages = ( pagesMapByCategory: Record< string, Pattern[] > ) => {
+const usePatternPages = (
+	pagesMapByCategory: Record< string, Pattern[] >,
+	categories: Category[],
+	dotcomPatterns: Pattern[]
+) => {
 	const [ searchParams, setSearchParams ] = useSearchParams();
+	let pageCategoriesInOrder;
+	let pageSlugs: string[] = [];
+	let pages: ( Pattern | null )[] = [];
+	let pagesToShow: any[] = [];
 	const page_slugs = searchParams.get( 'page_slugs' );
+	const custom_pages = searchParams.get( 'custom_pages' );
 
-	const pageSlugs = page_slugs !== null ? page_slugs.split( ',' ).filter( Boolean ) : INITIAL_PAGES;
+	if ( page_slugs !== null ) {
+		pageSlugs = page_slugs.split( ',' ).filter( Boolean );
+	}
 
-	const pages = pageSlugs.map( ( slug ) => pagesMapByCategory[ slug ]?.[ 0 ] ).filter( Boolean );
+	// eslint-disable-next-line prefer-const
+	pageCategoriesInOrder = useCategoriesOrder( categories, ORDERED_PATTERN_PAGES_CATEGORIES );
 
-	const setPageSlugs = ( slugs: string[] ) => {
-		setSearchParams(
-			( currentSearchParams ) => {
-				if ( slugs.length === 0 ) {
-					currentSearchParams.set( 'page_slugs', '' );
-				} else {
-					currentSearchParams.set( 'page_slugs', slugs.join( ',' ) );
-				}
+	// Pairs of page title and pattern id can be passed in the URL from AI flow.
+	if ( custom_pages !== null ) {
+		const mockedPages: CustomPageTitle[] = JSON.parse( custom_pages ) || [];
```

Comment:
Suggestion: Add `try/catch` here in case the JSON format is invalid.

Reply:
Done!